page-job-listing.php

```diff
@@ -0,0 +1,443 @@
+$salaryErrorClass = $salaryErrorMin = $salaryErrorMax = "";
+
+if (get_query_var('role_type')) {
+    $job_rôle_id = get_query_var('role_type');
+    if (is_numeric($job_rôle_id)) {
+        $job_rôle_id = intval($job_rôle_id);
+        if (term_exists($job_rôle_id, 'role_type')) {
+            $selected_job_role_id = $job_rôle_id;
+        }
+    }
+}
+if (get_query_var('job_region')) {
+    $job_region_id = get_query_var('job_region');
+    if (is_numeric($job_region_id)) {
+        $job_region_id = intval($job_region_id);
+        if (term_exists($job_region_id, 'job_region')) {
+            $selected_job_region_id = $job_region_id;
+        }
+    }
+}
+if (get_query_var('min_salary')) {
+    $min_salary_id = get_query_var('min_salary');
+    if (is_numeric($min_salary_id)) {
+        $selected_job_min_salary_id = intval($min_salary_id);
+    }
+}
+if (get_query_var('max_salary')) {
+    $max_salary_id = get_query_var('max_salary');
+    if (is_numeric($max_salary_id)) {
+        $selected_job_max_salary_id = intval($max_salary_id);
+    }
+}
+
+if (get_query_var('page_size')) {
+    $page_size_get = get_query_var('page_size');
+    if (is_numeric($page_size_get)) {
+        $page_size = intval($page_size_get);
+    }
+}
+
+
+
+$dropdown_html_role = wp_dropdown_categories(
+    array(
+        'name' => 'role_type',
+        'id' => 'job-filter-role',
+        'class' => 'govuk-select',
+        'taxonomy' => 'role_type',
+        'show_option_all' => 'Select option',
+        'orderby' => 'name',
+        'echo' => 0,
+        'hide_if_empty' => 1,
+        'selected' => $selected_job_role_id
+    )
+);
+
+foreach (get_terms(array( 'taxonomy' => 'job_region')) as $region) {
+    if (strtoupper($region->name) == "NATIONAL") {
+        $national_term_ID = $region->term_id;
+        break;
+    }
+}
+$dropdown_html_region = wp_dropdown_categories(
+    array(
+        'name' => 'job_region',
+        'id' => 'job-filter-region',
+        'class' => 'govuk-select',
+        'taxonomy' => 'job_region',
+        'show_option_all' => 'Select option',
+        'orderby' => 'name',
+        'echo' => 0,
+        'hide_if_empty' => 1,
+        'selected' => $selected_job_region_id,
+        'exclude' => $national_term_ID
+    )
+);
+
+if ($selected_job_max_salary_id != "0" && $selected_job_max_salary_id < $selected_job_min_salary_id ) {
+    $salaryError = true;
+    $salaryErrorClass = "govuk-select--error";
+    $salaryErrorMin = "Minimum salary cannot be higher than maximum salary.";
+    $salaryErrorMax = "Maximum salary cannot be lower than minimum salary.";
+}
+
+$selected0 = $selected20000 = $selected30000 = $selected40000 = $selected50000 = $selected60000 = $selected70000 = $selected80000 = $selected90000 = $selected100000 = "";
+$salarySelected = "selected$selected_job_min_salary_id";
+$$salarySelected = "selected";
+$dropdown_html_min_salary =
+"
+<select class='govuk-select $salaryErrorClass' name='min_salary' id='job-filter-min-salary'>
```

Comment:
could be changed template part to remove duplication

Reply:
I've turned this into a function